Library/Homebrew/exceptions.rb

```diff
@@ -251,7 +251,7 @@ def dump
       puts issues.map { |i| "#{i["title"]} #{i["html_url"]}" }.join("\n")
     end
 
-    if MacOS.version >= "10.11"
+    if OS::Mac.prerelease?
       require "cmd/doctor"
       opoo Checks.new.check_for_unsupported_osx
     end
```

Comment:
You mean require `cmd/doctor` and run `Checks#check_for_unsupported_osx` unconditionally after a failed build? I guess I agree that the micro-optimization done here is not needed after a possibly long build and it's even cleaner to leave the check entirely to the `check_*` methods.